x-pack/plugins/monitoring/server/config.ts

```diff
@@ -4,92 +4,24 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 import { schema, TypeOf } from '@kbn/config-schema';
+import {
+  configSchema as elasticsearchConfigSchema,
+  ElasticsearchConfig,
+  // eslint-disable-next-line @kbn/eslint/no-restricted-paths
+} from '../../../../src/core/server/elasticsearch';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { ElasticsearchConfigType } from '../../../../src/core/server/elasticsearch/elasticsearch_config';
```

Comment:
@restrry As we 'know' that these are used by external consumers, should we expose them publicly, or are these `no-restricted-paths` disabling fine?

Reply:
Deep imports are unnecessary here.
It can be generated locally with `TypeOf<...>`
```typescript
import { config, ElasticsearchConfig } from '../../../../src/core/server/';

const elasticsearchConfigSchema = config.elasticsearch.schema;
type ElasticsearchConfigType = TypeOf<typeof elasticsearchConfigSchema>;
```